### ISSUE TITLE
Sanity Studio won't start – Cannot read property 'sanity' of undefined

### ISSUE THREAD

**Issue by vonba** [sanity-io/sanity#2454]
NOTE: I am getting the same symptoms as this issue (link below), but supposedly it was resolved and to do with factors that don't seem to apply to my case, from what I can tell. I updated all packages to current versions immediately before the issue started.
https://github.com/sanity-io/sanity/issues/2190

**Describe the bug**

When starting Sanity, after installing the latest versions of all packages as per today (see full list below) I am getting the following error:

`Cannot read property 'sanity' of undefined`

The error (as in the issue linked to above) seems to be in GlobalStyle.js, L20:
`var color = theme.sanity.color.base;`

```
Stack:
TypeError: 
    at http://localhost:3333/static/js/app.bundle.js:342103:21
    at Ne (http://localhost:3333/static/js/app.bundle.js:71079:14925)
    at Ne (http://localhost:3333/static/js/app.bundle.js:71079:14698)
    at e.t.createStyles (http://localhost:3333/static/js/app.bundle.js:71079:23087)
    at e.t.renderStyles (http://localhost:3333/static/js/app.bundle.js:71079:23326)
    at h (http://localhost:3333/static/js/app.bundle.js:71079:24669)
    at http://localhost:3333/static/js/app.bundle.js:71079:24485
    at commitHookEffectListMount (http://localhost:3333/static/js/vendor.bundle.js:22889:26)
    at commitLifeCycles (http://localhost:3333/static/js/vendor.bundle.js:22945:9)
    at commitLayoutEffects (http://localhost:3333/static/js/vendor.bundle.js:25961:7)
```

**To Reproduce**

Install the versions listed below and start sanity.

**Expected behavior**

Sanity starting without issues.

**Which versions of Sanity are you using?**

These are the dependencies from my package.json:
```
    "@sanity/base": "^2.10.0",
    "@sanity/components": "^2.2.6",
    "@sanity/core": "^2.10.0",
    "@sanity/default-layout": "^2.10.0",
    "@sanity/default-login": "^2.8.0",
    "@sanity/desk-tool": "^2.10.0",
    "@sanity/google-maps-input": "^2.10.0",
    "@sanity/structure": "^2.10.0",
    "@sanity/vision": "^2.10.0",
    "prop-types": "^15.7",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-icons": "^4.2.0"
```

**What operating system are you using?**

MacOS Sierra

**Which versions of Node.js / npm are you running?**

6.14.11
v14.16.0

**Additional context**

The issue linked to above mentioned a problem to do with conflicting dependencies to Styled Components. I don't seem to be able to detect that issue in my setup.

**Comment by damky** [comment]
I'm getting a similar error.
`Cannot read property 'blue' of undefined

Stack:

TypeError: 
    at http://localhost:3333/static/js/app.bundle.js:121657:28
    at Array.reduce (<anonymous>)
    at Object.<anonymous> (http://localhost:3333/static/js/app.bundle.js:121655:33)
    at __webpack_require__ (http://localhost:3333/static/js/vendor.bundle.js:713:30)
    at fn (http://localhost:3333/static/js/vendor.bundle.js:118:20)
    at Object.<anonymous> (http://localhost:3333/static/js/app.bundle.js:121616:16)
    at __webpack_require__ (http://localhost:3333/static/js/vendor.bundle.js:713:30)
    at fn (http://localhost:3333/static/js/vendor.bundle.js:118:20)
    at Object.<anonymous> (http://localhost:3333/static/js/app.bundle.js:121592:18)
    at __webpack_require__ (http://localhost:3333/static/js/vendor.bundle.js:713:30)

(Your browsers Developer Tools may contain more info)`

**Comment by edkeohane** [comment]
This looks like it might be related to a build issue I also have on the most recent version.
https://github.com/sanity-io/sanity/issues/2459

**Comment by bjoerge** [comment]
I suspect this relates to `sanity upgrade` using yarn to install dependencies. Could you try removing node_modules and running `npm install` again?

**Comment by vonba** [comment]
> I suspect this relates to `sanity upgrade` using yarn to install dependencies. Could you try removing node_modules and running `npm install` again?

I already tried the steps you suggested (removing node_modules and running npm install).

I should mention that I my case the way I upgraded was to change my old package.json dependencies (which were locked to older package versions) to the newest versions. Did I miss a step by not running `sanity upgrade`?

**Comment by readeral** [comment]
> I suspect this relates to `sanity upgrade` using yarn to install dependencies. Could you try removing node_modules and running `npm install` again?

@bjoerge I've just experienced this issue too, is Yarn unable to be used for Sanity now? We're using yarn workspaces with Sanity as a sub-directory of a whole project and am not particularly inclined to change all our projects over to NPM workspaces!

**Comment by geoffreyjball** [comment]
Resolved in v2.19.0.
